tests/agent/test_event_handler.py

```diff
@@ -20,76 +46,152 @@
-        datasource_names_to_asset_names={"Data Source name": {"Asset name"}},
+class TestEventHandlerRegistry:
+    @pytest.mark.parametrize(
+        "version",
+        [
+            "0",
+            "1234567890",  # Version that doesn't exist already in the map
+        ],
     )
-    correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
-    handler = EventHandler(context=mock_context)
+    def test_register_event_action(self, version: str):
+        register_event_action(version, DummyEvent, DummyAction)
+        assert _EVENT_ACTION_MAP[version][DummyEvent.__name__] == DummyAction
 
-    handler.handle_event(event=event, id=correlation_id)
+        del _EVENT_ACTION_MAP[version][DummyEvent.__name__]  # Cleanup
 
-    action.assert_called_with(context=mock_context)
-    action().run.assert_called_with(event=event, id=correlation_id)
+    def test_register_event_action_already_registered(self):
+        register_event_action("0", DummyEvent, DummyAction)
+        with pytest.raises(ValueError):
+            register_event_action("0", DummyEvent, DummyAction)
 
+        del _EVENT_ACTION_MAP["0"][DummyEvent.__name__]  # Cleanup
 
-def test_event_handler_handles_draft_config_event(mock_context, mocker: MockerFixture):
-    action = mocker.patch(
-        "great_expectations_cloud.agent.event_handler.DraftDatasourceConfigAction"
-    )
-    event = DraftDatasourceConfigEvent(config_id=uuid4())
-    correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
-    handler = EventHandler(context=mock_context)
+    def test_event_handler_gets_correct_event_action(self, mock_context):
+        register_event_action("0", DummyEvent, DummyAction)
+        handler = EventHandler(context=mock_context)
+
+        assert type(handler.get_event_action(DummyEvent)) == DummyAction  # type: ignore[arg-type]  # Dummy event only used in testing
 
-    handler.handle_event(event=event, id=correlation_id)
+        del _EVENT_ACTION_MAP["0"][DummyEvent.__name__]
 
-    action.assert_called_with(context=mock_context)
-    action.return_value.run.assert_called_with(event=event, id=correlation_id)
+    @pytest.mark.parametrize(
+        "version, expected",
+        [
+            ("0.0.1", "0"),
+            ("1.0.1", "1"),
+            ("2.0.1", "2"),
+            ("1.1alpha1", "1"),
+            ("10.0.1", "10"),
+        ],
+    )
+    def test__get_major_version(self, version: str, expected: str):
+        assert _get_major_version(version) == expected
+
+    def test__get_major_version_raises_on_invalid_version(self):
+        with pytest.raises(InvalidVersionError):
+            with warnings.catch_warnings():
+                # Filter Deprecation warnings about LegacyVersion
+                warnings.filterwarnings(
+                    "ignore",
+                    message="Creating a LegacyVersion has been deprecated and will be removed in the next major release",
+                )
+                _get_major_version("invalid_version")
```

Comment:
Thank you for drilling down here, and only checking for the `InvalidVersionError` by filtering the warning :bow: